packages/react-native-web/src/exports/Image/types.js

```diff
@@ -102,17 +103,26 @@ export type ImageStyle = {
   tintColor?: ColorValue
 };
 
+export type SourceState = {
+  status: 'IDLE' | 'LOADING' | 'LOADED' | 'ERRORED',
+  source: ImageSource
+};
+
+export type ImageLoadingProps = {|
+  onLoad?: (e: { nativeEvent: ImageResult }) => void,
+  onLoadStart?: () => void,
+  onLoadEnd?: () => void,
+  onError?: ({ nativeEvent: { error: string } }) => void,
+  onProgress?: (e: any) => void
+|};
```

Comment:
These are used in the new `useSource` hook to limit flow type errors 
(By defining function input/output, we prevent the type checker to try and infer these types, because it couldn't)